### PR TITLE
Delete account

### PR DESCRIPTION
**Pull request by kristinyim**
User clicks on delete account link, opens up double confirmation, on click their account is deleted and they're taken to a signed out blank timetable.

Found a bug where the "isSigningUp" flag stays true when users create their acc and then open the account settings module so that the delete acc and other features don't appear. Should fix this later but need to get this feature out for now.

Also, we should find a better way to change the URL. Appending a tag and then clicking on it was the solution for Google and Facebook Signup so I just went with it but we could start using React Router?

image
image

### CODE REVIEW

File: student/urls.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
   url(r'^user/logout/$', 'django.contrib.auth.views.logout', {'next_page': '/'}),
   url(r'^unsubscribe/(?P<id>[\w.@+-]+)/(?P<token>[\w.:\-_=]+)/$', student.views.unsubscribe),
   url(r'^user/settings/$', student.views.UserView.as_view()),
+  url(r'^user/settings/delete/$', student.views.UserView.as_view()),
```
Comment (felixzhuologist) [inline student/urls.py]: why add the separate endpoint here? I think it's more RESTful to just user `user/settings`

File: static/js/redux/reducers/user_info_reducer.jsx [hunk 2]
```diff
@@ -51,6 +52,8 @@ const userInfo = (state = initialState, action) => {
       return Object.assign({}, state, { isVisible: true });
     case ActionTypes.SET_SETTINGS_MODAL_HIDDEN:
       return Object.assign({}, state, { isVisible: false });
+    case ActionTypes.DELETED_ACCOUNT:
```
Comment (felixzhuologist) [inline static/js/redux/reducers/user_info_reducer.jsx]: should we set data to `{ isLoggedIn: false}` here/do something to remove user info from the redux state? or does that happen automatically when you navigate to `user/logout`?

Comment (kristinyim) [reply static/js/redux/reducers/user_info_reducer.jsx]: when you navigate to `user/logout`, the entire redux state is cleared

**Comment by felixzhuologist** [comment]
build is failing because a node module is missing from the package.json - can you double check that the diff for that is what you expect?

**Comment by kristinyim** [comment]
Cherry picked in #1119

**Pull request closed without merging**